src/lib/ImpToLLVM.hs

```diff
@@ -451,26 +451,42 @@ compileInstr instr = case instr of
       -- if upcasting to unsigned int, use zext instruction
       (L.IntegerType _,    Scalar Word64Type)             -> x `zeroExtendTo` dt
       (L.IntegerType bits, Scalar Word32Type) | bits < 32 -> x `zeroExtendTo` dt
-      _ -> case (sxt, sdt) of
-       (L.IntegerType _, L.IntegerType _) -> x `asIntWidth` dt
-       (L.FloatingPointType fpt, L.FloatingPointType fpt') -> case compare fpt fpt' of
-         LT -> emitInstr dt $ L.FPExt x dt []
-         EQ -> return x
-         GT -> emitInstr dt $ L.FPTrunc x dt []
-       (L.FloatingPointType _, L.IntegerType _) -> emitInstr dt $ L.FPToSI x dt []
-       (L.IntegerType _, L.FloatingPointType _) -> emitInstr dt $ L.SIToFP x dt []
+      _ -> case (getIType ix, sdt) of
+        -- if upcasting from unsigned int, use zext instruction
+        (Scalar Word32Type, L.IntegerType bits) | bits > 32 -> x `zeroExtendTo` dt
+        (Scalar Word8Type,  L.IntegerType bits) | bits > 8  -> x `zeroExtendTo` dt
```

Comment:
Those seem new? Do we have any tests for them?

Reply:
Yes; the QuickCheck property will sometimes pick `Word8Lit` inputs to cast to `Int32Type` or `Int64Type`, and thus exercise this code path.